include/zephyr/arch/riscv/csr.h

```diff
@@ -233,4 +233,40 @@
 				: "memory");			\
 })
 
+/* uhartid -- user hardware thread id, seems to be not yet defined in toolchain */
+#define	uarthid		0x014
+
+#ifdef CONFIG_RISCV_KERNEL_IN_USER_MODE
+	/* zephyr runs in user mode */
+
+	/* inline assembly defines */
+	#define XSTATUS_IEN			USTATUS_IEN
+	#define XSTATUS_DEF_RESTORE		USTATUS_DEF_RESTORE
+	#define XSTATUS				0x000
+
+	/* register definition for assembly */
+	#define xhartid				uarthid
+	#define xscratch			uscratch
+	#define xstatus				ustatus
+	#define xepc				uepc
+	#define xcause				ucause
+	#define xret				uret
```

Comment:
As I remember the user-level Interrupts extension (N) was dropped from the RV spec - so I'm wondering if there were some changes about it or it's just for some old HW support?

Reply:
extension N seems to be dropped. there are devices that still support it. additionally the mechanism is same valid for supervisor mode. so the scheme can be extended to also support supervisor mode. nevertheless I have a device with N extension at hand, which works with this fix.